src/stacks-hierarchy/Root_TabNavigatorStack/Root_TabNavigatorStack.tsx

```diff
@@ -68,13 +84,27 @@ export const Root_TabNavigatorStack = ({navigation}: Props) => {
         );
       }
     }
+
+    if (
+      !notificationPermissionOnboarded &&
+      pushNotificationsEnabled &&
+      validFareContracts.length > 0 &&
+      notificationStatus !== 'granted' &&
+      locationWhenInUsePermissionOnboarded
+    ) {
+      InteractionManager.runAfterInteractions(() =>
+        navigation.navigate('Root_NotificationPermissionScreen'),
+      );
+    }
```

Comment:
Would something like this be a little clearer?
```tsx
const shouldOnboardNotifications =
  !notificationPermissionOnboarded &&
  pushNotificationsEnabled &&
  validFareContracts.length > 0 &&
  notificationStatus !== 'granted' &&
  locationWhenInUsePermissionOnboarded;

useEffect(() => {
  if (shouldOnboardNotifications) {
    InteractionManager.runAfterInteractions(() =>
      navigation.navigate('Root_NotificationPermissionScreen'),
    );
  }  
}, [shouldOnboardNotifications]);
```

Reply:
Yes that would be clearer, will make the change

Reply:
With the latest commit I belive it should stay like it is. @marius-at-atb can take a last look maybe?

Reply:
Thinking about it, I guess it could be a little clearer when splitting up like @gorandalum suggests. However just fine as is as well :+1: